docker-compose/docker-compose.yaml

```diff
@@ -102,7 +154,9 @@ services:
       - sourcegraph
     restart: always
     depends_on:
-      sourcegraph-frontend-internal:
```

Comment:
Why did this change end up being necessary? Removing the `sourcegraph-frontend-internal` means that we're going to restart both frontend containers at the same time during upgrdaes. That could be more disruptive if there's a problem bringing up the frontend service, running migrations, or any of its unofficial dependencies (gitserver etc.) are slow to start.

With the dependency on `sourcegraph-frontend-internal`, we keep the old public-facing frontend running until the internal frontend is healthy, and then swap over, to reduce the impact on users. It's unexpected that adding a migrator would require a change to this setup.

Reply:
Ultimately Dave and I didn't figure out why this was happening but when the `sourcegraph-frontend` was waiting for `sourcegraph-frontend-internal` it would never come up. `-internal` would prompt saying it's available on whatever port but the `frontend` container would never start up despite having a `restart: always` configuration which would eventually result in the container getting marked as unhealthy and tests failing. It was happening both locally and in the testing environment.

Reply:
Hm - I think what's happening is that now the `sourcegraph-frontend` container is failing its healthcheck (and restarting?) before the migrator starts. Both those containers start up at virtually the same moment. Once the migrator container begins running, the frontend healthcheck seems to notice that.

If I increase the interval or retries count on the `sourcegraph-frontend` healthcheck, `sourcegraph-frontend-internal` successfully starts up. Not great, though. Maybe we are just stuck with removing this dependency until we can upgrade docker-compose.

@efritz - does my theory sound plausible?

Reply:
@danieldides - With the `sourcegraph-frontend-internal` dependency removed, a failed migration now causes downtime. I was reluctant to tweak the interval / retries count but it may be a better option, compared to affecting customers. What do you think?

@efritz - It might help! But I'm not certain, and theoretically this problem will go away when we upgrade docker-compose anyway.